### PR TITLE
atlas: fix urls and shared libs

### PR DESCRIPTION
**Pull request by davydden**
without this patch, both url download `html`'s on macOS instead of tarbals.

tested on macOS with gcc6.1

@alalazo ping.

- [x] fix missing `$ nm -g libsatlas.dylib | grep _dggsvd_` which should be there.

It looks like `Lapack@3.6.0` deprecated `dggsvd` http://www.netlib.org/lapack/explore-html/dir_15e3bf8ae7662f502dfe98208b074c77.html , however `Trilinos` still needs it, but hopefully will fix it soon https://github.com/trilinos/Trilinos/issues/480

### CODE REVIEW

**Comment by alalazo** [comment]
@davydden In both cases I get this in the downloaded file : 
```html
<html><head>
<title>SourceForge</title>
<!-- <script src="/js/jquery.com/jquery-1.11.0.min.js"></script> -->
<script src="//code.jquery.com/jquery-1.11.0.min.js"></script>
<script src="//sourceforge.net/js/mirrors.js"></script>
<script src="/js/sf.js"></script>
<script>
var DR_loc = DR_parse_hash_url();
if (DR_loc) {
    DR_sf_main(DR_loc);
} else {
    window.location.href = 'http://sourceforge.net/home.html';
}
</script>
</head><body>
<noscript>
We're sorry -- the Sourceforge site is currently in Disaster Recovery mode, and currently requires
the use of javascript to function.  Please check back later.
</noscript>
</body></html>
```
so I don't know if the change in `url` is required. I am fine with the change in `lapack` version if `atlas` supports it.

**Comment by davydden** [comment]
@alalazo website is down (about an hour or so), please check later.

**Comment by davydden** [comment]
[macports](https://trac.macports.org/browser/trunk/dports/math/atlas/Portfile) have `3.6.0`, so it must be supported.

**Comment by alalazo** [comment]
@davydden Sorry, I thought the thing I posted WAS the html you were retrieving :smile:

**Comment by davydden** [comment]
nah, it was something else as after i changed url's i was able to download this morning.

**Comment by davydden** [comment]
@alalazo i updated Atlas PR quite a bit, pls give it a second look.

@tgamblin ready for review.

File: var/spack/repos/builtin/packages/atlas/package.py [hunk 1]
```diff
@@ -39,9 +39,11 @@ class Atlas(Package):
 
     version('3.10.2', 'a4e21f343dec8f22e7415e339f09f6da',
             url='https://sourceforge.net/projects/math-atlas/files/Stable/3.10.2/atlas3.10.2.tar.bz2', preferred=True)
+    # not all packages (e.g. Trilinos@12.6.3) stoped using depricated in 3.6.0
+    # Lapack rutines. Stick with 3.5.0 untill this is fixed.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/atlas/package.py]: typos: stopped, deprecated, routines, until

**Comment by davydden** [comment]
@tgamblin: unless  @alalazo  has further comments about this PR, i think it is ready.

**Comment by alalazo** [comment]
@davydden The only concern I have is about tests like the one here. We start to have a bunch of them (and this is good) but they are actually hard-coded in the packages and they require at the moment a bit too much code (at least in my opinion) in `package.py`. I think we reached a point in which it would be beneficial to think how to automate this before adding more and more tests.

So... I definitely don't want to stop this PR from being merged just to point out a potential issue for future PRs.

**Comment by davydden** [comment]
@alalazo i promise not to add more for now :smile: I create an issue to track this https://github.com/LLNL/spack/issues/1296

**Comment by davydden** [comment]
successfully tested on Ubuntu 14.04 with gcc 6.1.0.

@tgamblin this one is ready for merging.

**Comment by davydden** [comment]
@becker33 ping :smile:

**Comment by becker33** [comment]
I think we'd like to avoid having anything other than the package.py file live in the spack repo if possible. Is including the tests in the git repo the only way we can test atlas? If these files are available online, they can be retrieved as a resource.

**Comment by davydden** [comment]
@becker33 those aren't available online. I can of course get rid of those, if this is crucial.

**Comment by becker33** [comment]
Yes, please remove them. PR #1186 will increase our testing flexibility, is it possible that will allow you to run sufficient tests?

**Comment by becker33** [comment]
Hold that, had a quick discussion with @tgamblin and we're okay with doing the tests that way.

**Pull request merged**